src/lib/onlyText.ts

```diff
@@ -22,9 +21,14 @@ const onlyText = (children: ReactNode | ReactNode[]): string => {
   return Children.toArray(children).reduce((text: string, child: ReactNode): string => {
     let newText = '';
 
-    if (isValidElement(child) && hasChildren(child)) {
+    const isValidElementResult = isValidElement<{ children?: React.ReactNode | React.ReactNode[] }>(
+      child,
+    );
+    const hasChildren = isValidElementResult && 'children' in child.props;
```

Comment:
can we keep the same funcion call here? `hasChildren(child)`

Reply:
`hasChildren()` calls inside `isValidElement()` again

https://github.com/fernandopasik/react-children-utilities/blob/2e762942d15cf4c09acccc9ffa243a8179b64f85/src/lib/hasChildren.ts#L4-L7

we don't need to call `isValidElement()` twice.

Reply:
May be we need refactor `hasChildren` to:

```ts
 const hasChildren = (element: ReactElement<any>): element is ReactElement<{ children: ReactNode | ReactNode[] }> => Boolean(element.props.children); 
```

But it's major change.

Reply:
No, `hasChildren` calls isValid for a good reason, that also validates that is actually an element and not a child value